code/modules/clothing/spacesuits/rig/rig.dm

```diff
@@ -231,6 +231,24 @@
 			update_airtight(piece, 0) // Unseal
 	update_icon(1)
 
+/obj/item/weapon/rig/proc/cut_suit()
+	offline = 2
+	canremove = 1
+	for(var/obj/item/piece in list(helmet,boots,gloves,chest))
+		if(!piece) continue
+		piece.icon_state = "[suit_state]"
+		if(airtight)
+			update_airtight(piece, 0) // Unseal
+		piece.canremove = 1
+		if(istype(loc, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = loc
+			H.drop_from_inventory(piece)

```

Comment:
You could instead add a `force` parameter to `/obj/item/weapon/rig/proc/toggle_piece()` (Line ~760), such that instead of having to do all this inventory management, you can simply tell the proc to ignore stuff like battery checks?